politeiad/backend/gitbe/anchors.go

```diff
@@ -0,0 +1,338 @@
+}
+
+var (
+	regexCommitHash           = regexp.MustCompile(`^commit\s+(\S+)`)
+	regexCommitDate           = regexp.MustCompile(`^Date:\s+(.+)`)
+	anchorConfirmationPattern = fmt.Sprintf(`\s*%s\s+(\S+)`, markerAnchorConfirmation)
+	regexAnchorConfirmation   = regexp.MustCompile(anchorConfirmationPattern)
+	anchorPattern             = fmt.Sprintf(`\s*%s\s+(\S+)`, markerAnchor)
+	regexAnchor               = regexp.MustCompile(anchorPattern)
+)
+
+const (
+	gitDateTemplate = "Mon Jan 2 15:04:05 2006 -0700"
+)
+
+// extractCommit takes a slice of a git log and parses the next commit into a GitCommit struct
+func extractCommit(logSlice []string) (*GitCommit, int, error) {
+	var commit GitCommit
+
+	// Make sure we're at the start of a new commit
+	firstLine := logSlice[0]
+	if !regexCommitHash.MatchString(firstLine) {
+		return nil, 0, fmt.Errorf("Error parsing git log. Commit expected, found %q instead", firstLine)
+	}
+	commit.Hash = regexCommitHash.FindStringSubmatch(logSlice[0])[1]
+
+	// Skip the next line, which has the commit author
+
+	dateLine := logSlice[2]
+	if !regexCommitDate.MatchString(dateLine) {
+		return nil, 0, fmt.Errorf("Error parsing git log. Date expected, found %q instead", dateLine)
+	}
+	dateStr := regexCommitDate.FindStringSubmatch(logSlice[2])[1]
+	commitTime, err := time.Parse(gitDateTemplate, dateStr)
+	if err != nil {
+		return nil, 0, fmt.Errorf("Error parsing git log. Unable to parse date: %v", err)
+	}
+	commit.Time = commitTime.Unix()
+
+	// The first three lines are the commit hash, the author, and the date.
+	// The fourth is a blank line. Start accumulating the message at the 5th line.
+	// Append message lines until the start of the next commit is found.
+	for _, line := range logSlice[4:] {
+		if regexCommitHash.MatchString(line) {
+			break
+		}
+
+		commit.Message = append(commit.Message, line)
+	}
+
+	// In total, we used 4 lines initially, plus the number of lines in the message.
+	return &commit, len(commit.Message) + 4, nil
+}
+
+// getCommitsFromLog parses the git log into an array of GitCommit structs
+// It sends GitCommits into a channel, so that callers do not have to wait for the
+// entire log to be parsed.
+func (g *gitBackEnd) getCommitsFromLog(quit <-chan struct{}) <-chan *GitCommit {
+	// Create a channel to send commits
+	c := make(chan *GitCommit)
+
+	// Get the git log
+	gitLog, err := g.gitLog(g.vetted)
+	if err != nil {
+		c <- &GitCommit{Error: err}
+	}
+
+	// Parse the log into GitCommit structs for easier processing
+	// Quit early if a message comes from caller through the quit channel
+	go func() {
+		defer close(c)
+		currLine := 0
+		for currLine < len(gitLog) {
+			commit, linesUsed, err := extractCommit(gitLog[currLine:])
+			if err != nil {
+				c <- &GitCommit{Error: err}
+			}
+			currLine = currLine + linesUsed
+			select {
+			case <-quit:
+				return
+			case c <- commit:
+			}
+		}
+	}()
+
+	return c
+}
+
+// Some helper functions to navigate git commit message bodies
```

Comment:
You have to start these with the function name so that godoc knows what to do with it.